src/onnxflow/common/build.py

```diff
@@ -385,13 +365,28 @@ def save(self):
         else:
             state_dict["quantization_samples"] = False
 
+        return state_dict
+
+    def save_yaml(self, state_dict: Dict):
         with open(
             state_file(self.cache_dir, self.config.build_name), "w", encoding="utf8"
         ) as outfile:
             yaml.dump(state_dict, outfile)
 
+    def save(self):
+        self.prepare_file_system()
 
-def load_state(cache_dir=DEFAULT_CACHE_DIR, build_name=None, state_path=None) -> State:
+        state_dict = self.prepare_state_dict()
+
+        self.save_yaml(state_dict)
+
+
+def load_state(
+    cache_dir=DEFAULT_CACHE_DIR,
+    build_name=None,
+    state_path=None,
+    state_type: Type = State,
```

Comment:
This is the key change - libraries built on top of onnxflow can now supply their own State class. For example, `groqflow` will supply a `class GroqState(onnxflow.build.State)`.

Reply:
I will need a tutorial :)